src/pip/_internal/commands/search.py

```diff
@@ -140,6 +140,7 @@ def print_results(hits, name_column_width=None, terminal_width=None):
             write_output(line)
             if name in installed_packages:
                 dist = get_distribution(name)
+                assert dist is not None
```

Comment:
This whole module is using the distribution-related functions in a very inprecise way. I’ll submit a standalone refactoring PR after this is accepted